packages/core/admin/admin/src/content-manager/pages/ListView/components/BulkActionButtons/index.js

```diff
@@ -69,16 +77,79 @@ const confirmDialogsPropTypes = {
  * ConfirmDialogPublishAll
  * -----------------------------------------------------------------------------------------------*/
 
-const ConfirmDialogPublishAll = ({ isOpen, onToggleDialog, isConfirmButtonLoading, onConfirm }) => {
+export const ConfirmDialogPublishAll = ({
+  isOpen,
+  onToggleDialog,
+  isConfirmButtonLoading,
+  onConfirm,
+}) => {
   const { formatMessage } = useIntl();
+  const { get } = useFetchClient();
+  const { selectedEntries } = useTableContext();
+  const toggleNotification = useNotification();
+  const { formatAPIError } = useAPIErrorHandler(getTrad);
+  const {
+    contentType: { uid: slug },
+  } = useSelector(listViewDomain());
+
+  const {
+    data: countDraftRelations,
+    isLoading,
+    isError,
+  } = useQuery(
+    ['content-manager', 'draft-relations', slug, selectedEntries],
+    async () => {
+      /**
+       * TODO: refactor the API call when the param serializer will be ready
+       */
+      const {
+        data: { data },
+      } = await get(
+        `/content-manager/collection-types/${slug}/actions/countManyEntriesDraftRelations?${stringify(
+          { ids: selectedEntries },
+          { encode: false }
+        )}`
+      );
+
+      return data;
+    },
+    {
+      enabled: isOpen,
+      onError(error) {
+        toggleNotification({ type: 'warning', message: formatAPIError(error) });
+      },
+    }
+  );
+
+  if (isError) {
+    return null;
+  }
 
   return (
     <ConfirmBulkActionDialog
-      isOpen={isOpen}
+      isOpen={isOpen && !isLoading && !isError}
       onToggleDialog={onToggleDialog}
       dialogBody={
         <>
           <Typography id="confirm-description" textAlign="center">
+            {countDraftRelations > 0 &&
+              formatMessage(
+                {
+                  id: getTrad(`popUpwarning.warning.bulk-has-draft-relations.message`),
+                  defaultMessage:
+                    '<b>{count} {count, plural, one { relation } other { relations } } out of {entities} { entities, plural, one { entry } other { entries } } {count, plural, one { is } other { are } }</b> not published yet and might lead to unexpected behavior. ',
+                },
+                {
+                  // eslint-disable-next-line react/no-unstable-nested-components
```

Comment:
I don't think disabling eslint is necessary here, you can extract a simple component like we do [here](https://github.com/strapi/strapi/blob/4522fcab25fb1dce1591acc477b52322cdd9b14f/packages/core/admin/admin/src/content-manager/pages/ListView/components/BulkActionButtons/SelectedEntriesModal/index.js#L68)

Reply:
ok thanks

Reply:
done